src/libexec/zmrestoreblob

```diff
@@ -159,4 +168,29 @@ sub get_latest_backup_dir {
     my ($backup_path) = @_;
     my $latest_backup_dir = ( sort { -M $a <=> -M $b } glob($backup_path) )[0];
     return $latest_backup_dir;
+}
+
+sub check_msg_in_shared_blobs {
+    my ( $msg_id, $missing_blob ) = @_;
+    my $grid = $mailbox_id % 100;
+    $grid = ( $grid == 0 ) ? 100 : $grid;
+    my $blob_digest =
+`echo "select blob_digest from mail_item where blob_digest is not null and mailbox_id=${mailbox_id} and id=${msg_id};" | mysql -N mboxgroup${grid}`;
+    $blob_digest =~ s/[\r\n]//g;
+    $blob_digest = quotemeta($blob_digest);
+
+    my $zip              = Archive::Zip->new();
+    my @shared_blob_zips = glob( $latest_backup_path . '/shared_blobs/*' );
+    my $msg_match_flag   = 0;
+    foreach my $zip_file (@shared_blob_zips) {
```

Comment:
Given the duplication of code/logic in the foreach zipfile blocks here and starting on line 124, I'd recommend making those calls to a function and avoiding the duplication.